packages/peregrine/lib/talons/Navigation/useNavigation.js

```diff
@@ -11,16 +12,27 @@ const ancestors = {
     MENU: null
 };
 
-export const useNavigation = () => {
+export const useNavigation = props => {
+    const { customerQuery } = props;
+
     // retrieve app state from context
     const [appState, { closeDrawer }] = useAppContext();
     const [catalogState, { actions: catalogActions }] = useCatalogContext();
-    const [, { getUserDetails }] = useUserContext();
+    const [{ isSignedIn }, { actions: userActions }] = useUserContext();
+    const [getUserDetails, { data: getCustomerData }] = useLazyQuery(
+        customerQuery,
+        { fetchPolicy: 'network-only' }
```

Comment:
This is necessary until we investigate why `apolloClient.resetStore` wasn't clearing browser persistence.

Reply:
Add a comment in the code please :)